### ISSUE TITLE
xr-hyper does not work for \autoref and \hyperref

### ISSUE THREAD

**Issue by septatrix** [latex3/hyperref#156]
@davidcarlisle I did not find any bug report address on CTAN or in the docs and am not sure whether the issue is with xr/xr-hyper or hyperref itself. Therefore please excuse posting here and pinging.

<hr>

When linking between documents the aforementioned commands `\autoref` and `\hyperref` do not work while `\ref` and `\nameref` work without problems. They link to the wrong file (e.g. `cross-doc-b.pdf{}{}{}` - note the brackets) and have the wrong color (cyan instead of greenish-blue).

**Comment by septatrix** [comment]
`\pageref` also seems to work without problems. I also attached two files as a (nearly) minimal working example.

```latex
% cross-doc-a.tex
\documentclass{article}
\usepackage{xr-hyper}
\usepackage{hyperref}
\externaldocument[B-]{cross-doc-b}% <- full or relative path

\begin{document}

This is a test for math.
\begin{equation}
    E=mc^2 \label{eq:1}
\end{equation}

This is a second test for math.
\begin{equation}
    r = \sqrt{x^2 + y^2} \label{eq:2}
\end{equation}

Also see \autoref{B-sec:main}: \nameref{B-sec:main} or No. \ref{B-sec:main} on page \pageref{B-sec:main}.

Also see \hyperref[{B-sec:main}]{
  \autoref*{B-sec:main}: \nameref*{B-sec:main} or No. \ref*{B-sec:main}  on page \pageref*{B-sec:main}
}.

\end{document}
```

```latex
% cross-doc-b.tex
% the layout options were used to check if the links refer to the correct position
\documentclass[landscape, a5paper]{article}
\usepackage{xr-hyper}
\usepackage{hyperref}
\externaldocument[A-]{cross-doc-a}% <- full or relative path

\begin{document}

\setcounter{equation}{5}

As was shown in Eq.~(\ref{A-eq:1}) is it ... or in Eq.~(\ref{A-eq:2}) is ...

\newpage

\section{Main}\label{sec:main}

Also links to the correct position

\newpage

\section{Epilogue}

\end{document}
```

**Comment by u-fischer** [comment]
looks ok here. Is your system up-to-date? And did you compile everything more than once?

**Comment by septatrix** [comment]
I ran this on an up to date version of Ubuntu 20.04. Both files were compiled more than once. Also the text itself does work - it is only the links which point to a wrong file. I opened the PDF with evince. However I can also try to compile this using a more recent docker image with potentially newer releases from CTAN and see if the problem still exists.

**Comment by septatrix** [comment]
The docker container would have taken hours to download and overleaf also did not seem to behave different. However I checked the version of all relevant packages using `\listfiles` and `xr-hyper`, `hyperref` are both up-to-date. Using a different engine also does not seem to make a difference.

The links are clickable and visible in texstudio, evince and okular. Chrome does not show the colored boxes (know bug on their side) and you are only able to click the links generated by `\autoref` and `\hyperref` but those still refer to the wrong file. Firefox neither shows the boxes nor are you able to follow the links. Therefore you would need to use one of the former viewers to inspect this bug.

<details>
<summary>Output of `\listfiles`</summary>

```
*File List*
 article.cls    2019/12/20 v1.4l Standard LaTeX document class
  size10.clo    2019/12/20 v1.4l Standard LaTeX file (size option)
xr-hyper.sty    2019/10/03 v6.1 eXternal References (DPC)
hyperref.sty    2020/01/14 v7.00d Hypertext links for LaTeX
 ltxcmds.sty    2019/12/15 v1.24 LaTeX kernel commands for general use (HO)
   iftex.sty    2019/11/07 v1.0c TeX engine tests
pdftexcmds.sty    2019/11/24 v0.31 Utility functions of pdfTeX for LuaTeX (HO)
infwarerr.sty    2019/12/03 v1.5 Providing info/warning/error messages (HO)
  keyval.sty    2014/10/28 v1.15 key=value parser (DPC)
kvsetkeys.sty    2019/12/15 v1.18 Key value parser (HO)
kvdefinekeys.sty    2019-12-19 v1.6 Define keys (HO)
pdfescape.sty    2019/12/09 v1.15 Implements pdfTeX's escape features (HO)
 hycolor.sty    2020-01-27 v1.10 Color options for hyperref/bookmark (HO)
letltxmacro.sty    2019/12/03 v1.6 Let assignment for LaTeX macros (HO)
 auxhook.sty    2019-12-17 v1.6 Hooks for auxiliary files (HO)
kvoptions.sty    2019/11/29 v3.13 Key value format for package options (HO)
  pd1enc.def    2020/01/14 v7.00d Hyperref: PDFDocEncoding definition (HO)
 intcalc.sty    2019/12/15 v1.3 Expandable calculations with integers (HO)
etexcmds.sty    2019/12/15 v1.7 Avoid name clashes with e-TeX commands (HO)
     url.sty    2013/09/16  ver 3.4  Verb mode for urls, etc.
  bitset.sty    2019/12/09 v1.3 Handle bit-vector datatype (HO)
bigintcalc.sty    2019/12/15 v1.5 Expandable calculations on big integers (HO)
atbegshi.sty    2019/12/05 v1.19 At begin shipout hook (HO)
 hluatex.def    2020/01/14 v7.00d Hyperref driver for luaTeX
   puenc.def    2020/01/14 v7.00d Hyperref: PDF Unicode definition (HO)
stringenc.sty    2019/11/29 v1.12 Convert strings between diff. encodings (HO)
atveryend.sty    2019-12-11 v1.11 Hooks at the very end of document (HO)
rerunfilecheck.sty    2019/12/05 v1.9 Rerun checks for auxiliary files (HO)
uniquecounter.sty    2019/12/15 v1.4 Provide unlimited unique counter (HO)
l3backend-pdfmode.def    2020-02-03 L3 backend support: PDF mode
  ts1cmr.fd    2019/12/16 v2.5j Standard LaTeX font definitions
 nameref.sty    2019/09/16 v2.46 Cross-referencing by name of section
refcount.sty    2019/12/15 v3.6 Data extraction from label references (HO)
gettitlestring.sty    2019/12/15 v1.6 Cleanup title references (HO)
cross-doc-a.out
cross-doc-a.out
 ***********
```

</details>

**Comment by u-fischer** [comment]
neither xr-hyper nor hyperref is up-to-date in your system. The current versions are 

~~~~
xr-hyper.sty    2020-05-15 v7.00e eXternal References (DPC)
hyperref.sty    2020-05-15 v7.00e Hypertext links for LaTeX
~~~~

**Comment by septatrix** [comment]
Oh you are right. I only checked the github releases and missed that there are newer versions on CTAN. Just downloaded those and everything seems to work fine. The problem was presumably the same you fixed in #140 so thanks for that! I do not know if there are still some problems causing #118 to be kept open but as this is fixed in the new version and a duplicate of said issue I will close this.
